File: Data Structures & Algorithms/max-water-container/submission-5.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let left = 0
        let right = heights.length - 1
        let maxArea = 0
        
        while (left <right){
            const area = (right - left)*Math.min(heights[left], heights[right])
            maxArea = Math.max(area, maxArea)

            if (heights[left] < heights[right]){
                left+=1
            }
            else right-=1
        }

        return maxArea
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-5.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left = 0
        total = 0
        seen = set()
        for right in range(len(s)):
            print(seen)
            while s[right] in seen:
                seen.remove(s[left])
                left+=1
            else:
                total = max(total, right-left+1)
                seen.add(s[right])
        return total

